tools/linker/MonoTouch.Tuner/PreserveSmartEnumConversionsSubStep.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright 2017 Xamarin Inc.
+
+using System;
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Linker;
+using Mono.Tuner;
+using MonoTouch.Tuner;
+
+namespace Xamarin.Linker.Steps
+{
+	public class PreserveSmartEnumConversionsSubStep : ExceptionalSubStep
+	{
+		Dictionary<TypeDefinition, Tuple<MethodDefinition, MethodDefinition>> cache;
+		protected override string Name { get; } = "Smart Enum Conversion Preserver";
+		protected override int ErrorCode { get; } = 2100;
+
+		public override SubStepTargets Targets {
+			get {
+				return SubStepTargets.Method | SubStepTargets.Property;
+			}
+		}
+
+		public override bool IsActiveFor (AssemblyDefinition assembly)
+		{
+			// we need to process all assemblies, because the functions we want to
+			// preserve are not necessarily in the assembly we're processing.
+			return true;
```

Comment:
Sounds heavy, processing seems done only when `[BindAs]` is seen so this could check if the type is referenced by the assembly, right ? That should remove processing the whole BCL and most of user code.

As for preserving other ~~functions~~ methods it should not matter as you use annotations (either linkskip or linksafe should be happy). Maybe that comment/code was meant for an earlier implementation ? (I can see how this could be a problem)